azure/scope/machinepool.go

```diff
@@ -484,6 +564,7 @@ func (m *MachinePoolScope) SetFailureMessage(v error) {
 }
 
 // SetFailureReason sets the AzureMachinePool status failure reason.
+// TODO: should this be capierrors.MachinePoolStatusFailure instead?
```

Comment:
Is this TODO going to be left for a later PR or revision?

Reply:
It's more of a question I had. I can remove the TODO but just wanted to get some clarification on if I'm understanding correctly.

Reply:
It won't change much in practice, they are both a string type, but yes IMO `capierrors.MachinePoolStatusFailure` would be more appropriate

Reply:
Should we change this to use `capierrors.MachinePoolStatusFailure` (or open an issue to refactor it afterward)?